compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -330,10 +340,8 @@ impl StructType {
     /// True if the given index is the same index as a generic type of this struct
     /// which is expected to be a numeric generic.
     /// This is needed because we infer type kinds in Noir and don't have extensive kind checking.
-    /// TODO(https://github.com/noir-lang/noir/issues/5156): This is outdated and we should remove this implicit searching for numeric generics
     pub fn generic_is_numeric(&self, index_of_generic: usize) -> bool {
```

Comment:
I believe this method can be entirely deleted

Reply:
It's still used in [the elaborator](https://github.com/noir-lang/noir/blob/d35baf19537dbe525942ec197564a6ec7c5f8254/compiler/noirc_frontend/src/elaborator/types.rs#L1718): would you prefer for it to be inlined there?

Reply:
Isn't it only used by the `find_numeric_generics` method?

Reply:
Yes

Reply:
Yeah so we should be deleting all usage of `find_numeric_generics` in this PR.